test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/KeepAliveTimeoutTests.cs

```diff
@@ -0,0 +1,223 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests
+{
+    public class KeepAliveTimeoutTests
+    {
+        private static readonly TimeSpan KeepAliveTimeout = TimeSpan.FromSeconds(1);
+        private const int LongDelay = 5000; // milliseconds
+        private const int ShortDelay = 250; // milliseconds
+
+        [Fact]
+        public async Task ConnectionClosedWhenKeepAliveTimeoutExpires()
```

Comment:
I don't think xunit will run tests in the same class in parallel by default. Since these are longer running tests, is there a way we can make these run concurrently?